src/auth/authOS.ts

```diff
@@ -0,0 +1,246 @@
+        authenticationPrompt,
+        accessControl: Keychain.ACCESS_CONTROL.BIOMETRY_CURRENT_SET_OR_DEVICE_PASSCODE,
+      })
+
+      if (!credentials) throw new Error('Failed to load secret')
+      return credentials.password
+    },
+    ...options,
+  })
+
+  return {
+    loadSecret: mutation.mutate,
+    ...mutation,
+  }
+}
+
+export const useResetSecret = (options?: UseMutationOptions<boolean, Error, {key: string}>) => {
+  const mutation = useMutation({
+    mutationFn: async ({key}: {key: string}) =>
+      Keychain.resetGenericPassword({
+        service: key,
+      }),
+    ...options,
+  })
+
+  return {
+    resetSecret: mutation.mutate,
+    ...mutation,
+  }
+}
+
+export const useAuthMethod = (storage: Storage, options?: UseQueryOptions<AuthMethodState, Error>) => {
+  const query = useQuery({
+    queryKey: ['authMethod'],
+    queryFn: () =>
+      Promise.resolve(authMethodKey)
+        .then(storage.getItem)
+        .then((storedAuthMethod) => (!isEmptyString(storedAuthMethod) ? storedAuthMethod : '""'))
+        .then(JSON.parse)
+        .then((parsedAuthMethod) => {
+          switch (parsedAuthMethod) {
+            case 'pin':
+              return {
+                isPIN: true,
+                isOS: false,
+                isNone: false,
+              } as const
+            case 'os':
+              return {
+                isPIN: false,
+                isOS: true,
+                isNone: false,
+              } as const
+            default:
+              return {
+                isPIN: false,
+                isOS: false,
+                isNone: true,
+              } as const
+          }
+        }),
+    ...options,
+  })
+
+  return {
+    authMethod: query.data,
+    ...query,
+  }
+}
+
+export const useSaveAuthMethod = (storage: Storage, options?: UseMutationOptions<void, Error, AuthMethod>) => {
+  const mutation = useMutationWithInvalidations({
+    mutationFn: (authMethod) => storage.setItem(authMethodKey, JSON.stringify(authMethod)),
+    invalidateQueries: [['authMethod']],
+    ...options,
+  })
+
+  return {
+    ...mutation,
+    saveAuthMethod: mutation.mutate,
+  }
+}
+
+export const useAuthOsAppKey = (storage: Storage, options?: UseQueryOptions<string, Error>) => {
+  const query = useQuery({
+    suspense: true,
+    queryKey: ['installationId'],
+    queryFn: () =>
+      storage
+        .getItem('/appSettings/installationId')
+        .then((data) => {
+          if (data != null) return data
+          throw new Error('Invalid state')
+        })
+        .then(JSON.parse),
+    ...options,
+  })
+
+  return query.data
+}
+
+export const useChangeAuthMethod = (storage: Storage, options?: UseMutationOptions<void, Error, AuthMethod>) => {
+  const mutation = useMutationWithInvalidations({
+    mutationFn: (authMethod) => storage.setItem(authMethodKey, JSON.stringify(authMethod)),
+    invalidateQueries: [['authMethod']],
+    ...options,
+  })
+
+  return mutation.mutate
```

Comment:
why not return the whole mutation? what about errors and loading?

Reply:
I copied to `useSaveAuthMethod` and forgot to delete it 🤦